lib/tests/test_core.c

```diff
@@ -101,15 +101,19 @@ test_probability_list_select(void)
     }
     {
         double probs[3] = { short_half, DBL_EPSILON / 8.0, 0.5 };
+#ifdef MSP_TEST_EXACT_FLOAT_COMPARISONS
```

Comment:
Didn't you say you didn't want conditional compiles cluttering up the code? :smiling_imp:
(looks fine to me, i'm just teasing you on `#ifdef NDEBUG` comments long ago)

Reply:
Well it was either that or delete the tests, and this is a good sanity check when we have proper maths on board. I guess we could do a custom macro, but comparing floats for exact equality is rarely a good idea anyway, so didn't seem worth it.